python/tests/test_account.py

```diff
@@ -442,32 +442,41 @@ def test_one_account_send_bcc_setting(self, acfactory, lp):
         assert self_addr not in ev[2]
         ev = ac1._evlogger.get_matching("DC_EVENT_DELETED_BLOB_FILE")
 
-    def test_prepare_file_with_unicode(self, acfactory, lp):
+    def test_send_file_twice_unicode_filename_mangling(self, tmpdir, acfactory, lp):
         ac1, ac2 = acfactory.get_two_online_accounts()
         chat = self.get_chat(ac1, ac2)
 
-        lp.sec("ac1: prepare and send attachment + text to ac2")
-        blobdir = ac1.get_blobdir()
-        basename = "somedäüta.txt"
-        p = os.path.join(blobdir, basename)
+        basename = "somedäüta.html.zip"
+        p = os.path.join(tmpdir.strpath, basename)
         with open(p, "w") as f:
             f.write("some data")
-        msg = Message.new_empty(ac1, "file")
-        msg.set_text("hello ä world")
-        msg.set_file(p)
-        message = chat.prepare_message(msg)
-        assert message.is_out_preparing()
-        assert message.text == "hello ä world"
-        chat.send_prepared(message)
 
-        lp.sec("ac2: receive message")
-        ev = ac2._evlogger.get_matching("DC_EVENT_INCOMING_MSG|DC_EVENT_MSGS_CHANGED")
-        assert ev[2] > const.DC_CHAT_ID_LAST_SPECIAL
-        msg = ac2.get_message_by_id(ev[1])
-        assert msg.text == "hello ä world"
+        def send_and_receive_message():
+            lp.sec("ac1: prepare and send attachment + text to ac2")
+            msg1 = Message.new_empty(ac1, "file")
+            msg1.set_text("withfile")
+            msg1.set_file(p)
+            message = chat.prepare_message(msg1)
```

Comment:
@r10s it is my understanding this is wrong: `p` at this point is a file which is **not** in the blobdir.  But when calling dc_prepare_msg() any files **must** be in the blobdir.  Is my understanding correct?

Reply:
yip, when calling dc_prepare_msg() the file must be placed in the blob directory for $reasons i cannot remember completely, was related to early-mime-generation (typically, this affects videos or recoding images and, in general we want to avoid that there are references in the database for external files)